clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/manifest/KubernetesKindRegistry.java

```diff
@@ -17,23 +17,27 @@
 package com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
-import java.util.Objects;
+import java.util.Map;
 import java.util.Optional;
-import java.util.function.Predicate;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Supplier;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import org.apache.commons.lang3.StringUtils;
 
 public class KubernetesKindRegistry {
-  private final List<KubernetesKind> values = Collections.synchronizedList(new ArrayList<>());
+  private final Map<KubernetesKind.ScopedKind, KubernetesKind> nameMap = new ConcurrentHashMap<>();
+  private final Map<KubernetesKind.ScopedKind, KubernetesKind> aliasMap = new ConcurrentHashMap<>();
```

Comment:
Is the reason we want a `ConcurrentHashMap` instead of a boring regular `HashMap` because (quoting from the internet) "If you have very highly concurrent application in which you may want to modify or read key value in different threads then it’s ideal to use Concurrent Hashmap."?

Reply:
I used a `ConcurrentHashMap` because we might be reading from multiple threads at a time (though we can only be updating one thread at a time because the update methods are `synchronized`) and might be reading from a thread while another thread is performing an update.  Using a `ConcurrentHashMap` guarantees we'll see consistent results in that case, whereas a regular `HashMap` would require manually adding thread-safety on top.

Reply:
:+1: :+1: :+1: thanks!